src/reporter.ts

```diff
@@ -241,56 +241,47 @@ class RPReporter implements Reporter {
   }
 
   onTestBegin(test: TestResp): void {
-    //create suite
-    const suiteHasParent = test.parent.parent?._isDescribe;
-    const suiteTitle = suiteHasParent ? test.parent.parent?.title : test.parent.title;
-    if (!this.findTestItem(this.suites, suiteTitle)) {
-      const codeRef = getCodeRef(test, TEST_ITEM_TYPES.SUITE);
+    const suitesOrder: string[] = [];
+    const createSuitesOrder = (suite: any) => {
+      if (!suite?._isDescribe) {
+        return;
+      }
+      suitesOrder.push(suite.title);
+      createSuitesOrder(suite.parent);
+    };
+    createSuitesOrder(test.parent);
+
+    //create suites
+    for (let i = 0; i < suitesOrder.length; i++) {
+      const index = suitesOrder.length - i - 1;
+      if (this.findTestItem(this.suites, suitesOrder[index])) {
```

Comment:
in case of you want to iterate in desc ordering better using `for (let i = arr.length - 1; i >= 0; i--) {}` and avoid unnecessary variables like `const index = suitesOrder.length - i - 1;`